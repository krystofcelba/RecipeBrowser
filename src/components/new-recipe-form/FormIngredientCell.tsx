import React from 'react';
import { View, StyleSheet, TextInput } from 'react-native';
import { Cell } from 'react-native-tableview-simple';

import { StylesConstants } from '../../assets/constants';
import { NewIngredient } from '../../redux/reducers/ui';
import Input from './Input';
import i18n from '../../assets/i18n';

interface Props {
  ingredient: NewIngredient;
  onChange: (ingredient: NewIngredient) => void;
}

const FormIngredientCell = ({ ingredient, onChange }: Props) => (
  <Cell
    cellContentView={
      <View style={styles.container}>
        <Input
          style={styles.nameInput}
          placeholder={i18n.t('name')}
          value={ingredient.name}
          onChangeText={name => onChange({ ...ingredient, name })}
        />
        <Input
          style={styles.amountInput}
          placeholder={i18n.t('amount')}
          value={ingredient.amount}
          onChangeText={amount => onChange({ ...ingredient, amount })}
        />
        <Input
          style={styles.unitInput}
          placeholder={i18n.t('unit')}
          value={ingredient.unit}
          onChangeText={unit => onChange({ ...ingredient, unit })}
        />
      </View>
    }
  />
);

const styles = StyleSheet.create({
  container: { width: '100%', paddingVertical: 10, flexDirection: 'row' },
  nameInput: {
    width: '60%',
  },
  amountInput: {
    width: '20%',
    textAlign: 'center',
  },
  unitInput: {
    width: '20%',
    textAlign: 'right',
  },
});

export default FormIngredientCell;
