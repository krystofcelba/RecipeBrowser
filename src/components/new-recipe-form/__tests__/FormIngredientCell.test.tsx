import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import FormIngredientCell from '../FormIngredientCell';

describe('FormIngredientCell component', () => {
  it('renders as expected', () => {
    const wrapper = shallow(
      <FormIngredientCell ingredient={{ id: 1, name: 'test', amount: '0', unit: 'kg' }} onChange={() => {}} />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('call onChange as expected', () => {
    const onChangeSpy = jest.fn();
    const wrapper = shallow(
      <FormIngredientCell ingredient={{ id: 1, name: 'test', amount: '0', unit: 'kg' }} onChange={onChangeSpy} />,
    );
    shallow(wrapper.prop('cellContentView'))
      .find('Input')
      .forEach(child => {
        child.simulate('changeText', 'a');
      });
    expect(onChangeSpy.mock.calls).toMatchSnapshot();
  });
});
