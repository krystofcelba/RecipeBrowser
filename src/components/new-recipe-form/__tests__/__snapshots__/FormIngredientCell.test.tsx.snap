// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormIngredientCell component call onChange as expected 1`] = `
Array [
  Array [
    Object {
      "amount": "0",
      "id": 1,
      "name": "a",
      "unit": "kg",
    },
  ],
  Array [
    Object {
      "amount": "a",
      "id": 1,
      "name": "test",
      "unit": "kg",
    },
  ],
  Array [
    Object {
      "amount": "0",
      "id": 1,
      "name": "test",
      "unit": "a",
    },
  ],
]
`;

exports[`FormIngredientCell component renders as expected 1`] = `
<Cell
  accessory={false}
  accessoryColor="#007AFF"
  allowFontScaling={true}
  backgroundColor="#FFF"
  cellAccessoryView={null}
  cellContentView={
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingVertical": 10,
          "width": "100%",
        }
      }
    >
      <Input
        onChangeText={[Function]}
        placeholder="Name"
        style={
          Object {
            "width": "60%",
          }
        }
        value="test"
      />
      <Input
        onChangeText={[Function]}
        placeholder="Amount"
        style={
          Object {
            "textAlign": "center",
            "width": "20%",
          }
        }
        value="0"
      />
      <Input
        onChangeText={[Function]}
        placeholder="Unit"
        style={
          Object {
            "textAlign": "right",
            "width": "20%",
          }
        }
        value="kg"
      />
    </View>
  }
  cellImageView={null}
  cellStyle="Basic"
  contentContainerStyle={Object {}}
  detail=""
  detailTextStyle={Object {}}
  disableImageResize={false}
  highlightActiveOpacity={0.8}
  highlightUnderlayColor="black"
  image={null}
  isDisabled={false}
  leftDetailColor="#007AFF"
  onHighlightRow={null}
  onPress={false}
  onUnHighlightRow={null}
  rightDetailColor="#8E8E93"
  subtitleColor="#000"
  subtitleTextStyle={Object {}}
  title=""
  titleTextColor="#000"
  titleTextStyle={Object {}}
  titleTextStyleDisabled={Object {}}
/>
`;
